### PR TITLE
Fix OSGi dependency definitions

### PR DESCRIPTION
**Pull request by Mailaender**
Otherwise, it crashes at runtime because `LoggerFactory` or `DataSet` is not found.

### CODE REVIEW

File: jhdf/build.gradle [hunk 1]
```diff
@@ -104,7 +104,8 @@ jar {
             'Bundle-Name': project.name,
             'Bundle-Vendor': 'James Mudd',
             'Bundle-Version': project.version,
-            'Export-Package': 'io.jhdf,io.jhdf.*',
+            'Export-Package': 'io.jhdf,io.jhdf.*,io.jhdf.api,io.jhdf.api.*',
+            'Require-Bundle': 'slf4j.api;bundle-version="1.7.36",org.apache.commons.lang3;bundle-version="3.13.0",com.ning.compress-lzf;bundle-version="1.1.2",lz4-java;bundle-version="1.8.0"',
```
Comment (Mailaender) [inline jhdf/build.gradle:108]: We may want to have version ranges here instead of hard-coded versions.

**Comment by jamesmudd** [comment]
Thanks for looking at this. Could the ranges be expanded, e.g. `'slf4j.api;bundle-version="[1.7.36,2.0.0)"` currently the versions of these dependencies get auto-updated for minor and patch versions.

Ideally maybe these headers could be generated in the build but that might be a bit more complex.

**Comment by Mailaender** [comment]
Sure. According to https://www.slf4j.org/download.html the 1.X branch is no longer developed. I also tried 1.7.30 which is what Eclipse ships, but I couldn't make it work.

**Comment by jamesmudd** [comment]
Actually I have reconsidered this. I don't think we should add version ranges. I have no setup to test combinations of dependencies, so I think just pinning to the specified versions is safest for now.

**Comment by Mailaender** [comment]
Removed all version ranges.

**Pull request merged**